### ISSUE TITLE
add cop for "global method"

### ISSUE THREAD

**Issue by grosser** [rubocop/rubocop#9734]
## feature request

We have a project were many people contribute that don't know ruby well, they often write methods that are global, I'd like to stop them
this also often happens in things like Rakefile or *.rake files

## Describe the solution you'd like

```ruby
class Foo
  def not_global <- Good
  end
end
def this_is_global <- Bad
end
```

## Describe alternatives you've considered

`git grep '^def ` works, but would be nice if it's in rubocop

**Comment by bbatsov** [comment]
The idea makes sense to me, although such a cop should likely be disabled by default as for many scripts top-level method definition are perfectly fine.
